core/src/test/scala/integration/kafka/api/CorruptedBrokersTest.scala

```diff
@@ -0,0 +1,117 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package integration.kafka.api
+
+import kafka.api.{IntegrationTestHarness, LeaderAndIsr}
+import kafka.log.Log.offsetFromFile
+import kafka.log.{LogSegment}
+import kafka.server.KafkaConfig
+import kafka.utils.TestUtils
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerConfig, ProducerRecord}
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.utils.Time
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+
+import java.util.Properties
+
+class CorruptedBrokersTest extends IntegrationTestHarness {
+  val brokerCount = 3
+  val controllerId = 0
+  val broker1Id = 1
+  val broker2Id = 2
+  val topic = "topic1"
+  val partition = 0
+  val topicPartition = new TopicPartition(topic, partition)
+
+  serverConfig.setProperty(KafkaConfig.LiDropCorruptedFilesEnableProp, true.toString)
+  serverConfig.setProperty(KafkaConfig.LiLeaderElectionOnCorruptionWaitMsProp, 30000.toString)
+  serverConfig.setProperty(KafkaConfig.ControlledShutdownEnableProp, false.toString)
+  serverConfig.setProperty(KafkaConfig.MinInSyncReplicasProp, 1.toString)
+  serverConfig.setProperty(KafkaConfig.UncleanLeaderElectionEnableProp, true.toString)
+
+  producerConfig.setProperty(ProducerConfig.LINGER_MS_CONFIG, 0.toString)
+  producerConfig.setProperty(ProducerConfig.BATCH_SIZE_CONFIG, 0.toString)
```

Comment:
This is also the way it is done in other Kafka unit tests, so it was done in keeping pattern. I can change them to string literals.
I generally prefer to use the original types over a string because it prevents problems like writing `"flase"` by accident, and the test code interpreting it in a slient manner and still passing (can cause tests to not break when they should). I agree that this is quite roundabout in this case, so I am not very insistent on it.

I verified that these configs actually take effect. Also, I found other classes who are doing it this way, not needing to override `setup()`.